src/DemoRunner.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Website;
+
+use Nette\Utils\Json;
+use Rector\Website\Entity\RectorRun;
+use Rector\Website\Process\RectorProcessRunner;
+use Rector\Website\Utils\FileDiffCleaner;
+use function Sentry\captureException;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Throwable;
+
+final class DemoRunner
+{
+    /**
+     * @var RectorProcessRunner
+     */
+    private $rectorProcessRunner;
+
+    /**
+     * @var FileDiffCleaner
+     */
+    private $fileDiffCleaner;
+
+    public function __construct(RectorProcessRunner $rectorProcessRunner, FileDiffCleaner $fileDiffCleaner)
+    {
+        $this->rectorProcessRunner = $rectorProcessRunner;
+        $this->fileDiffCleaner = $fileDiffCleaner;
+    }
+
+    public function runAndPopulateRunResult(RectorRun $rectorRun): void
+    {
+        $stopwatch = new Stopwatch();
+        $rectorProcessStopwatchEvent = $stopwatch->start('rector-process');
+
+        try {
+            $runResult = $this->rectorProcessRunner->run($rectorRun);
+            $fileDiff = $this->createFileDiff($runResult, $rectorRun);
+
+            $rectorRun->success($fileDiff, Json::encode($runResult), $rectorProcessStopwatchEvent);
+        } catch (Throwable $throwable) {
+            $rectorRun->fail($throwable->getMessage(), $rectorProcessStopwatchEvent);
```

Comment:
@JanMikes  Btw, this looks like `$rectorRun` is a service, not an object.
It would be better to use clear method naming e.g. `markAsFailed()` or create this object with a factory.